site-src/geps/gep-1058.md

```diff
@@ -0,0 +1,962 @@
+
+```yaml
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: gateway-a
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 80
+    hostname: *.example.com
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: gateway-b
+spec:
+  gatewayClassName: nadir-lb
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 80
+    hostname: subdomain.example.com
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-parent
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  - name: gateway-b
+    sectionName: http
+  rules:
+    - allowedRoutes:
+        kinds: 
+          - kind: HTTPRoute
+        namespaces:
+          from: Same
+    - backendRefs:
+      - name: default-svc
+        port: 8080
+status:
+  parents:
+  - controllerName: acme
+    parentRef:
+      name: gateway-a
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Successfully accepted
+  - controllerName: nadir
+    parentRef:
+      name: gateway-b
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Successfully accepted
+---
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: HTTPRoute
+metadata:
+  name: http-child
+spec:
+  parentRefs:
+  - name: gateway-a
+    sectionName: http
+  hostnames:
+    - testdomain.example.com
+  rules:
+    - backendRefs:
+      - name: foo-svc
+        port: 8080
+status:
+  parents:
+  - controllerName: acme
+    parentRef:
+      name: http-parent
+    conditions:
+    - type: Accepted
+      status: true
+      reason: Accepted
+      message: Accepted successfully
+    grandparents:
+      - parentRef:
+          name: gateway-a
+        conditions:
+        - type: Accepted
+          status: true
+          reason: Accepted
+          message: Successfully accepted
+  - controllerName: nadir
+    parentRef:
+      name: http-parent
+    conditions:
+    - type: Accepted
+      status: true
+      reason: GrandparentConfigError
+      message: Some configuration was not accepted in the attachment process. See grandparents for more details.
+    grandparents:
+      - parentRef:
+          name: gateway-b
+        conditions:
+        - type: Accepted
+          status: false
+          reason: HostnameMismatch
+          message: The route could not be accepted because of a hostname mismatch - `testdomain.example.com` does not match `subdomain.example.com`.
+```
+
+
+
+## Alternatives
+
+### Extending backendRef to allow targeting a Route
+
+This idea was to allow `backendRef` to, optionally, specify a route instead of a
+service. Some of the reasons for not taking this approach include:
+
+- It would require changes to `backendRefs`. No changes to `allowedRoutes` are
+  needed to use it for this. 
+- To minimize the changes to `backendRef`, it would require the owner of the
```

Comment:
Not necessarily - we could allow abstractions such as just selecting the namespace and leaving name blank, which would be functionally equivalent to allowNamespaces.

Reply:
+1, or we can have the backendRef match whatever we define in GAMMA. 

Leaving it blank is a bit problematic because in a namespace you may have children of multiple gateways. For example /admin may be delegated by foo.com and bar.com.  I don't think requiring hostname in child routes is ideal - and also not sufficient, I think it is possible for 2 different gateways in 2 namespaces to define 'foo.com' with different IPs - and still delegate.